containers/src/Data/IntMap/Internal.hs

```diff
@@ -2067,7 +2071,7 @@ merge g1 g2 f m1 m2 =
 --
 -- @since 0.5.9
 mergeA
-  :: (Applicative f)
+  :: forall f a b c . (Applicative f)
```

Comment:
Could you explain why we need this change, and the type signature below?

Reply:
Yeah, that seems like a surprising choice. FWIW, GHC seems to be going in the opposite direction of making that even more explicit.